dipy/core/tests/test_geometry.py

```diff
@@ -239,7 +254,42 @@ def test_compose_decompose_matrix():
                     assert_array_almost_equal(scale, sc)
 
 
+def test_perpendicular_directions():
+    num = 35
+    vector_v = sphere2cart(1, np.pi/4, np.pi/4)
+    pd = perpendicular_directions(vector_v, num=num, half=False)
+
+    # see if length of pd is equal to the number of intendend samples
+    assert_equal(num, len(pd))
+
+    # check if all directions are perpendicular to vector v
+    for d in pd:
+        cos_angle = np.dot(d, vector_v)
+        assert_almost_equal(cos_angle, 0)
+
+    # check if directions are sampled by multiples of 2*pi / num
+    delta_a = 2 * np.pi / num
+    for d in pd:
+        angle = np.arccos(np.dot(pd[0], d))
+        rest = angle % delta_a
+        if rest > delta_a * 0.99:  # To correct cases of negative error
+            rest = rest - delta_a
+        assert_almost_equal(rest, 0)
+
+    # check case of vector_v is aligned to x_axis
+    vector_v = [1., 0., 0.]
+    pd = perpendicular_directions(vector_v, num=num, half=False)
+    for d in pd:
```

Comment:
To reduce repetition in this test, you could write it as a loop:
    
    for vv in [sphere2cart(1, np.pi/4, np.pi/4), [1., 0., 0.]]:
        ...